### PR TITLE
[TS] LPS-153799

### PR DESCRIPTION
**Pull request by balazssk**
Hi Team,

Can you please review?
https://issues.liferay.com/browse/LPS-153799

Thanks,
Balázs

Motivation
=======
Custom field (expando) references in segments are not resolved if imported into a new instance
[Link to story or ticket](https://issues.liferay.com/browse/LPS-153799)

Proposed Solution
========
By replacing the expando field reference in segements with their StagedExpandoColumn's uuid for the time of the export, we can recalculate the the expando field's PK before importing it

Steps to Verify:
----------------
1. Create an expando field for Users
1. Export Expando Fields into a LAR and save it
1. Create a Segment using the new expando field
1. Export Segments into a LAR and save it
1. Delete the Segment
1. Delete the expando field
1. Import the expando field LAR
1. Import the Segment LAR

Change summary (optional):
---------------
* Updated Test to simulate importing into a new instance (removed the original expando column, so its import would change its PK)
* Implement a replacement logic for expando columns for the export-import phase so that the column is exported in an identifyable way and that identity is resolved during import, before the criteria is parsed

Checklist (optional):
========
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [?] My changes generate no new warnings - I usually use CI to decide this
- [x] I have added tests that prove my fix is effective or that my feature works
- [?] New and existing unit tests pass locally with my changes - New integration tests passes, I usually use CI to run the whole suit
- [x] Any dependent changes have been merged and published in downstream modules - Can't think of dependent modules

### CODE REVIEW

**Comment by cgoncas** [comment]
Hey @balazssk !!!

Could you follow the Play described here: https://liferay.atlassian.net/wiki/spaces/ENGTANGO/pages/1368786134/Play+Pull+Request+Reviews

Thanks!!!

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by balazssk** [comment]
Hi @cgoncas ,
Thanks for the playbook, I've updated the description to best match it.
I have a question regarding this `remember to push the branch to liferay-tango` is this something Support should also follow?

**Comment by cgoncas** [comment]
ci:test:tango

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/exportimport/content/processor/SegmentsEntryExportImportContentProcessor.java [hunk 1]
```diff
@@ -120,6 +132,91 @@ private String _replaceExportCriteriaReferences(
...
+			matcher.appendReplacement(
+				sb,
+				Matcher.quoteReplacement(
+					"customField\\/_" + uuid.replaceAll(" ", "_")));
+		}
+
+		if (sb != null) {
+			matcher.appendTail(sb);
+
+			content = sb.toString();
```
Comment (cgoncas) [inline modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/exportimport/content/processor/SegmentsEntryExportImportContentProcessor.java:169]: What do you think about return directly sb.toString()?

Comment (balazssk) [reply modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/exportimport/content/processor/SegmentsEntryExportImportContentProcessor.java:169]: that could work, I'm just a fan of single returns

**Comment by cgoncas** [comment]
ci:forward

**Pull request closed without merging**